mesonbuild/interpreter/interpreter.py

```diff
@@ -91,7 +91,7 @@ def stringifyUserArguments(args, quote=False):
         return '[%s]' % ', '.join([stringifyUserArguments(x, True) for x in args])
     elif isinstance(args, dict):
         return '{%s}' % ', '.join(['{} : {}'.format(stringifyUserArguments(k, True), stringifyUserArguments(v, True)) for k, v in args.items()])
-    elif isinstance(args, int):
+    elif type(args) == int: # isinstance will accept bools too
```

Comment:
you should use `is` not `==` here.

But it would be more idiomatic to do something like:
```python
elif isinstance(args, bool):
    pass  # we want this to fallthrough to the error case
elif isinstance(args, int):
    return str(args)
```

Reply:
Fair enough.